RestaurantReservation/Program.cs

```diff
@@ -1,6 +1,72 @@
 ﻿using RestaurantReservation.Db;
+using RestaurantReservation.Db.Models;
+using RestaurantReservation.Services;
 
-using (RestaurantReservationDbContext context = new())
+using var context = new RestaurantReservationDbContext();
+
+// Create services for each entity
+var customerService = new CustomerService(context);
+var restaurantService = new RestaurantService(context);
+var reservationService = new ReservationService(context);
+var orderItemService = new OrderItemService(context);
+var orderService = new OrderService(context);
+var tableService = new TableService(context);
+var employeeService = new EmployeeService(context);
+var menuItemService = new MenuItemService(context);
+
+// Create sample data
+var customer = new Customer { FirstName = "John", LastName = "Doe", Email = "john@example.com", PhoneNumber = "123-456-7890" };
+var restaurant = new Restaurant { RestaurantName = "Sample Restaurant", Address = "123 Main St", PhoneNumber = "987-654-3210", OpeningHours = "8:00 AM - 10:00 PM" };
+var reservation = new Reservation { CustomerId = 3, TableId = 3, RestaurantId = 1, ReservationDate = DateTime.Now, PartySize = 4 };
+var orderItem = new OrderItem { OrderId = 3, ItemId = 1, Quantity = 2 };
+var order = new Order { ReservationId = 3, EmployeeId = 1, OrderDate = DateTime.Now, TotalAmount = 50.0 };
```

Comment:
How do you know that this FK exists?
I think for the Dependent entity we can pass it to the Principal entity or we need to create the Principal first and then get its ID to use it with the Dependent

Reply:
I chose them from the data I seeded